### ISSUE TITLE
chanId not being set by ca_create_channel()

### ISSUE THREAD

**Issue by DKastelic** [wanglin86769/node-epics-ca#2]
I'm trying to run the examples from README.md, but whenever `pv.connect()` is called, the following error is thrown:

``` 
TypeError: Unexpected <type_41> * value, expected chanId
    at /home/dkastelic/node-epics-testing/node_modules/node-epics-ca/lib/channel.js:204:30
    at new Promise (<anonymous>)
    at Channel.connect (/home/dkastelic/node-epics-testing/node_modules/node-epics-ca/lib/channel.js:197:16)
    at /home/dkastelic/node-epics-testing/index.js:7:18
    at Object.<anonymous> (/home/dkastelic/node-epics-testing/index.js:14:3)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
 ```
 
I believe the issue is that `ca_create_channel()` saves `{}` to `chidPtr[0]` instead of a channel id, but it only becomes apparent when `ca_field_type()` recieves this value as an argument and throws an error.

I'm running both an instance of the EPICS example IOC and Node.js in wsl, and can successfully caget the PV I am using in the script from the console.

**Comment by wanglin86769** [comment]
Hi, I can reproduce the issue, thanks for reporting. 
I will try to fix it.

**Comment by wanglin86769** [comment]
node-epics-ca relies on koffi, the root cause is that koffi version 2.5.4 breaks the previously working code of node-epics-ca.

With the help of the author of koffi as follows, the issue has been fixed.
https://github.com/Koromix/koffi/issues/62

I have published the new version **node-epics-ca 0.2.1**, please try it.

**Comment by DKastelic** [comment]
Thank you, it seems to work now.

**Comment by DKastelic** [comment]
Hi, today I got the same error again when trying to manually close a channel opened with `CA.monitor()`:
``` 
/home/dkastelic/node-epics-server/node_modules/node-epics-ca/lib/channel.js:227
                    let errCode = ca_clear_subscription(this.monitorEventId);
                                  ^

TypeError: Unexpected <type_41> * value, expected evid
    at Timeout._onTimeout (/home/dkastelic/node-epics-server/node_modules/node-epics-ca/lib/channel.js:227:35)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7)
```

Simple example that reproduces the issue:
```js
const CA = require('node-epics-ca');
(async () => {
    let channel = await CA.monitor('calcExample', function(data) {
        console.log('Current:', data);
    });
    
    setTimeout(async function() {
        console.log("Closing channel manually.");
        await channel.disconnect();
    }, 2 * 1000);
    
    setTimeout(function() {
        console.log("Done!!!");
    }, 5 * 1000);
})()
```

I think I managed to fix it by modifying ca_create_subscription in the same way that ca_create_channel was modified:
from 
```js
const ca_create_subscription = libca.func('ca_create_subscription', 'int', ['int','ulong',chanId,'long',koffi.pointer(MonitorCallback),'pointer',koffi.out(pointer)]);
```
to
```js
const ca_create_subscription = libca.func('ca_create_subscription', 'int', ['int','ulong',chanId,'long',koffi.pointer(MonitorCallback),'void *',koffi.out(koffi.pointer(evid))]);
```

**Comment by wanglin86769** [comment]
Hi, thanks for your report as well as the solution.
I have published v0.2.2 with the solution you suggested.
I forgot to modify the "void *" part, but it does not seem to affect.

**Comment by DKastelic** [comment]
Thanks!
